Course/Course/Domain/Model/CourseDates.swift

```diff
@@ -14,41 +14,46 @@ public struct CourseDates {
     let hasEnded, learnerIsFullAccess: Bool
     let userTimezone: String?
     
-    var sortedDateToCourseDateBlockDict: [Date: [CourseDateBlock]] {
-        var dateToCourseDateBlockDict: [Date: [CourseDateBlock]] = [:]
-        var hasToday = false
-        let today = Date.today
-        
-        let calendar = Calendar.current
-        let todayComponents = calendar.dateComponents([.year, .month, .day], from: .today)
+    var sortedStatusToDateToCourseDateBlockDict: [CompletionStatus: [Date: [CourseDateBlock]]] {
+        var statusToDateToCourseDateBlockDict: [CompletionStatus: [Date: [CourseDateBlock]]] = [:]
+        var statusToCourseDateBlockDict: [CompletionStatus: [CourseDateBlock]] = [:]
         
         for block in courseDateBlocks {
             let date = block.date
-            let dateComponents = calendar.dateComponents([.year, .month, .day], from: date)
-            
-            if dateComponents == todayComponents {
-                hasToday = true
+            switch true {
```

Comment:
It's just an alternate approach to prevent having a lengthy series of if-else blocks. Each case independently assesses its own condition to build a completion status dictionary.